src/dodal/devices/tetramm.py

```diff
@@ -125,20 +128,29 @@ async def prepare(self, trigger_info: TriggerInfo):
         # trigger mode must be set first and on its own!
         await self._drv.trigger_mode.set(TetrammTrigger.EXT_TRIGGER)
 
+        # ensure trigger_info.number_of_events is iterable of numbers
+        total_triggers = (
+            sum(trigger_info.number_of_events)
+            if isinstance(trigger_info.number_of_events, list | tuple)
+            else int(trigger_info.number_of_events)
+        )
+        self._total_triggers = total_triggers
+        self._averaging_time = trigger_info.livetime / self._total_triggers
```

Comment:
Could: Can the averaging time just be a local variable since it's not used outside this method?

```suggestion
        averaging_time = trigger_info.livetime / self._total_triggers
```